scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -237,8 +238,8 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
                 }.associate { it.await() }
             }
         } finally {
-            storageDispatcher.close()
-            scanDispatcher.close()
+            STORAGE_DISPATCHER.close()
+            SCAN_DISPATCHER.close()
```

Comment:
I didn't think about closing the dispatchers before, but this will close the dispatchers too early, as soon as the first of the two scanners has finished. The only clean solution I can currently think of is to make the dispatchers class properties and make the `LocalScanner` `Closeable`, and then close it from the `Scanner` as soon as everything is done. If `scanPackages` is called again after closing the `LocalScanner` it should either fail or recreate the dispatchers (thread safe).

Reply:
Do you know whether it's even required to close the dispatchers? So far it seems to me the only thing closing does is shutting down the executors, which in turn just seems to [avoid more working being scheduled](https://stackoverflow.com/a/16742217/1127485).

Reply:
Yes, otherwise the process would keep running: https://stackoverflow.com/a/29999187/639421
It might be an option to use daemon threads, though.